Sources/BraveWallet/Crypto/AssetIconView.swift

```diff
@@ -103,3 +103,25 @@ struct AssetIconView_Previews: PreviewProvider {
   }
 }
 #endif
+
+struct NFTIconView: View {
+  
+  /// NFT token
+  var token: BraveWallet.BlockchainToken
+  /// Network for token
+  var network: BraveWallet.NetworkInfo
+  /// NFT image url from metadata
+  var url: URL?
+  
+  var body: some View {
+    WebImageReader(url: url) { image, isFinished in
+      if let image = image {
+        Image(uiImage: image)
+          .resizable()
+          .aspectRatio(contentMode: .fit)
+      } else {
+        AssetIconView(token: token, network: network)
+      }
```

Comment:
do you think we need a loader?

Reply:
should we just use monogram as a fullback view instead using AssetIconView which again checks token's symbol and logo to load?

Reply:
I think more of a concern in #6361 when we actually show images when we are fetching metadata for NFTs. Will need to fetch from URL, decode from base64, maybe other formats. Just did basic support for now.